src/functionalTest/groovy/org/gradle/crypto/checksum/CacheChecksumSpec.groovy

```diff
@@ -0,0 +1,126 @@
+package org.gradle.crypto.checksum
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.Specification
+import spock.lang.TempDir
+
+class CacheChecksumSpec extends Specification {
+
+    @TempDir File testProjectDir
+    File localBuildCacheDirectory
+    File buildDirRun1
+    File buildDirRun2
+
+    def setup() {
+        localBuildCacheDirectory = new File(testProjectDir, 'local-cache')
+        buildDirRun1 = new File(testProjectDir, 'run1')
+        setupBuildDir(buildDirRun1)
+        buildDirRun2 = new File(testProjectDir, 'run2')
+        setupBuildDir(buildDirRun2)
+
+        new File(testProjectDir,'input').mkdir()
+        new File(testProjectDir,'input/foo.txt') << 'foo'
+        new File(testProjectDir,'input/bar.txt') << 'bar'
+        new File(testProjectDir,'input/subdir').mkdirs()
+        new File(testProjectDir,'input/subdir/sub-foo.txt') << 'sub-foo'
+        new File(testProjectDir,'input/baz.txt') << 'baz'
+    }
+
+    def 'can use build cache on #gradleVersion'() {
```

Comment:
This doesn't test for build-cache relocatability. Is that intended?

Reply:
The builds are running from 2 different folders. Am I missing something?

Reply:
Ha! No, it is me that missed this :+1: